test/controllers/products_controller_test.rb

```diff
@@ -0,0 +1,150 @@
+require "test_helper"
+
+describe ProductsController do
+  
+  describe "#index action" do
+    it "responds with success when there is at least one Product saved" do
+      new_product = products(:p1)
+      
+      get products_path
+      
+      must_respond_with :success
+      expect(Product.count).must_be :>, 0
+    end
+    
+    it "responds with success when there are products to show " do
+      get products_path
+      
+      must_respond_with :success
+      expect(Product.count).must_equal 5
+    end
+  end
```

Comment:
These tests are actually testing the same condition. (The arrange in the 1st test isn't actually changing the db.) An interesting edge case would be to delete all the products and make sure the page still responds the way you want it to.